src/main/java/org/cirdles/topsoil/dataset/DatasetResourceManager.java

```diff
@@ -0,0 +1,108 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.cirdles.topsoil.app.dataset.reader.DatasetReader;
+import org.cirdles.topsoil.app.dataset.reader.TSVDatasetReader;
+
+/**
+ *
+ * @author parizotclement
+ */
+public class DatasetResourceManager {
+
+    private final List<DatasetResource> datasetResources;
+
+    public DatasetResourceManager(Path basePath) {
+
+        datasetResources = getInitialDatasetResources(basePath);
+
+        try {
+            Files.createDirectories(basePath);
+        } catch (IOException ex) {
+            Logger.getLogger(DatasetResourceManager.class.getName()).log(Level.SEVERE, null, ex);
+        }
+    }
+
+    public void addDatasetResources(DatasetResource datasetResource) {
+        datasetResources.add(datasetResource);
+    }
+
+    public void removeDatasetResources(DatasetResource datasetResource) {
+        datasetResources.remove(datasetResource);
+    }
+
+    public List<DatasetResource> getDatasetResources() {
+        return datasetResources;
+    }
+
+    private List<DatasetResource> getInitialDatasetResources(Path basePath) {
+        List<DatasetResource> resources = new ArrayList();
+
+        resources.addAll(getDatasetsForPath(basePath.resolve("open")));
+        resources.addAll(getDatasetsForPath(basePath.resolve("closed")));
+
+        return resources;
+    }
+
+     private List<DatasetResource> getDatasetsForPath(Path path) {
+        List<DatasetResource> resources = new ArrayList<>();
+        DatasetReader datasetReader = new TSVDatasetReader(true);
+
+        try {
+            // create the path if it does not yet exist
+            Files.createDirectories(path);
+
+            Files.walk(path, 1).forEach(datasetPath -> {
+                try {
+                    if (Files.isDirectory(datasetPath)) {
+                        return;
+                    }
+
+                    SimpleDataset dataset = (SimpleDataset) datasetReader.read(datasetPath);
+                    resources.add(DatasetResourceFactory.getDatasetResource(dataset, datasetPath, path.getParent()));
+
+                    dataset.setName(datasetPath.getFileName().toString());
+                } catch (IOException ex) {
+                    Logger.getLogger(DatasetResourceManager.class.getName()).log(Level.SEVERE, null, ex);
+                }
+            });
+        } catch (IOException ex) {
+            Logger.getLogger(DatasetResourceManager.class.getName()).log(Level.SEVERE, null, ex);
+        }
+
+        return resources;
+    }
+
+    public Optional<DatasetResource> getDatasetResource(Dataset dataset) {
+        Optional<DatasetResource> datasetResource = Optional.empty();
+        for (int i = 0 ; i < datasetResources.size(); i++) {
+            if (datasetResources.get(i).getDataset() == dataset) {
+                datasetResource = Optional.of(datasetResources.get(i));
+                break;
+            }
+        }
+        return datasetResource;
+    }
+}
```

Comment:
As I've already said, I feel it is more necessary to have this class instead of just a plain list because its purpose is both to create the directories and to retrieve the datasets at the application's initialization. Yes we could maybe do that in the `TopsoilMainWindow`, but as you've said it is already overloaded, and I think it would be out of place to do it in the `DatasetResourceFactory`. 
What do you think would be better ?